cookbooks/composition_dependent_phase_transition/doc/composition_dependent_phase_transition.md

```diff
@@ -0,0 +1,100 @@
+| --------- | --------- | --------- | --------- | --------- | --------- |
+|grt -> CaPv + grt | 560 | 60 | 4.0 | 6.7 | - |
+|grt -> il| 660 | 5 | 4.0 | 5.6 | $T < 1662 K$ |
+|il -> pv| 660 | 5 | -3.1 | 5.97 | $T < 1662 K$ |
+|grt -> pv | 660 | 5 | 1.3 | 11.7| $T > 1662 K$ |
+```
+
+The three compositions are then compiled as a mixture of olivine and garnet.
+The details of this approach are discussed in {cite:t}`arredondo:2016`
+
+
+## The input file
+
+First, we illustrate the setup with pyrolite composition.
+The related entries in the prm file are listed below.
+A visualization of the result is generated following the setup in Section&nbsp;{ref}`sec:cookbooks:visualizing_phase_diagram` (  {numref}`fig:pyrolite_density`).
+
+```{literalinclude} pyrolite_material_model.prm
+```
+
+Each of the phase transitions is parameterized by their respective depth, temperature, clapeyron slope, and width.
+Moreover, two other parameters, namely the "Phase transition temperature lower limits" and the "Phase transition temperature upper limits" limit a PT to a range of temperatures, which is necessary for garnet phase transitions in the MTZ  ({numref}`fig:pyrolite_density`, the 1st, 3rd red lines and the 2nd orange line).
+
+Besides, we'll also need to make sure that the phase transitions don't intersect with each other, thus we'll need to separate the rw -> pv + mw into two parts (the 2nd red line and the 1st orange line).
+We can see this by dividing the deeper PTs into a cooler group (red lines) and a warmer group (orange lines).
+
+Last, in each of the temperature groups, we need to prescribe the PTs in a top-down manner: from shallower to deeper depth.
+
+For the careful measurement, one needs to take during this process, it's recommended to generate one phase diagram for every composition following the instructions in {ref}`sec:cookbooks:visualizing_phase_diagram` before running any dynamic models.
+
+You can find the whole parameter file for this phase diagram in
+[cookbooks/layered_phase_transition/pyrolite_phase_diagram.prm](https://www.github.com/geodynamics/aspect/blob/main/cookbooks/layered_phase_transition/pyrolite_phase_diagram.prm)
+
+```{figure-md} fig:pyrolite_density
+<img src="pyrolite_density_marked.png" style="width:90.0%" />
+
+An example of the pyrolite composition. Red and orange marks on phase transitions demonstrate the usage of the "temperature limits": Red - low-T PTs, Orange - high-T PTs.
+```
+
+## Results
+
+A complex setup is illustrated with a 60-degree dipping slab penetrating into the lower mantle.
+In this example, an oceanic crust locates in the upper 7.5 km of the slab overlying a 37.5 km thick harzburgite layer, with the pyrolite composition assigned to the rest of the mantle ({numref}`fig:layered_initial`).
+The resulting density structure is shown in figure ({numref}`fig:layered_result` a).
+
+In addition to densities, a set of rheology parameters could also be assigned with the same interfaces.
```

Comment:
Should the above "are assigned" instead of "could also be assigned"?

Reply:
yes, changed